pepagent/pepagent.py

```diff
@@ -304,31 +402,45 @@ def _get_namespace_proj_anno(self, namespace: str = None) -> dict:
 
         return res_dict
 
-    def run_sql_search_single(self, sql_query: str, *argv) -> list:
+    def run_sql_fetchone(self, sql_query: str, argv=()) -> list:
         """
         Fetching one result by providing sql query and arguments
         :param str sql_query: sql string that has to run
         :param argv: arguments that has to be added to sql query
         :return: set of query result
         """
+        # coerce argv to tuple if not
+        if isinstance(argv, list):
+            argv = tuple(argv)
+        elif not isinstance(argv, tuple):
+            argv = (argv,)
         cursor = self.postgresConnection.cursor()
         try:
             cursor.execute(sql_query, argv)
             output_result = cursor.fetchone()
-            cursor.close()
-            return list(output_result)
+
+            # must run check here since None is not iterable.
+            if output_result is not None:
+                return list(output_result)
+            else:
+                return None
         except psycopg2.Error as e:
             _LOGGER.error(f"Error occurred while running query: {e}")
         finally:
             cursor.close()
 
-    def run_sql_search_all(self, sql_query: str, *argv) -> list:
+    def run_sql_fetchall(self, sql_query: str, argv=()) -> list:
```

Comment:
Why did you change *args to args as set?

Reply:
Yeah, so `psycopg` wants a tuple, so I thought we should be passing a tuple as well to the `run_sql_fetchall` function... I am not sure if this **truly** is best. A variable-length list of unnamed args after the `sql_query` arg seemed dicey to me. More granular control over a tuple. I guess, why not just be explicit in the way we pass in the parameters for the query.

I am open to reverting, though. Maybe @nsheff has a preference.

Reply:
Ok. Reading more into this, I think your approach was better. After looking through this thread on SO I am more partial to `*args`: https://stackoverflow.com/questions/33542959/why-use-packed-args-kwargs-instead-of-passing-list-dict

I'll revert back today and make sure all tests still pass.

Reply:
Fixed.